### PR TITLE
Move join and left mesages to lua

### PR DESCRIPTION
**Pull request by xunto**
When I was reading #4454, I found out that messages that is written when player join and leave the game is defined in cpp. I don't see any reason for that, so I moved it to builtin. Now you can modify it by editing builtin lua.

I'm waiting for suggestion about how to make it customizable by mods(there are several ways). My opinion is that we shouldn't define any default, so if somebody would want those messages, they should be defined in mod.

### CODE REVIEW

**Comment by est31** [comment]
:+1: looks good.

I don't think it needs to be exposed via the modding api. It will make it much much easier for server owners to customize the message now that's it in builtin.

**Comment by nerzhul** [comment]
okay for me

File: builtin/game/misc.lua [hunk 1]
```diff
@@ -87,11 +87,19 @@ end
...
+	player_list[player_name] = player
+	core.chat_send_all("*** " .. player_name .. " joined the game.")
 end)
 
-core.register_on_leaveplayer(function(player)
-	player_list[player:get_player_name()] = nil
+core.register_on_leaveplayer(function(player, timed_out)
+	local player_name = player:get_player_name()
+	player_list[player_name] = nil
+	local announcement = "*** " ..  player_name .. " left the game.";
```
Comment (HybridDog) [inline builtin/game/misc.lua]: you do not need a semicolon here

**Comment by HybridDog** [comment]
Please give the messages a grey colour.

**Comment by xunto** [comment]
@HybridDog I think that is should be done in separate pull request, because objective of this PR is to move messages it to builtin and to make them customizable.

Not everybody would like to have grey colour. It should be discussed and not here.

**Comment by HybridDog** [comment]
ok, sorry

**Comment by xunto** [comment]
@HybridDog lol, you shouldn't delete your message, you are breaking the disscussion. And there is nothing to be sorry about.

**Comment by HybridDog** [comment]
It'sn't even my idea https://github.com/minetest/minetest/issues/4454#issuecomment-241071522

**Pull request merged**